File: app/routers/tickets.py
from fastapi import APIRouter, Depends, status
from sqlalchemy.orm import Session

from .. import models, schemas
from ..database import get_db
from ..deps import get_current_user
from ..services import ticket_service

router = APIRouter(prefix="/api/tickets", tags=["tickets"])


@router.post("/hold", response_model=schemas.TicketHoldResponse)
async def hold_ticket(
    data: schemas.TicketHoldRequest,
    db: Session = Depends(get_db),
    user: models.User = Depends(get_current_user),
):
    return await ticket_service.hold_ticket(db, data, user)


@router.post("/{ticket_id}/release", status_code=status.HTTP_204_NO_CONTENT)
async def release_ticket(
    ticket_id: str,
    db: Session = Depends(get_db),
    user: models.User = Depends(get_current_user),
):
    await ticket_service.release_ticket_hold(db, ticket_id)

File: src/routers/tickets.js
'use strict';
const express = require('express');
const ticketService = require('../services/ticketService');
const { TicketHoldRequest } = require('../schemas');
const { validateBody } = require('../middleware/validate');
const { getCurrentUser } = require('../middleware/auth');
const { asyncHandler } = require('../utils/errors');

const router = express.Router();

router.post(
  '/hold',
  getCurrentUser,
  validateBody(TicketHoldRequest),
  asyncHandler(async (req, res) => {
    const result = await ticketService.holdTicket(req.validated, req.user);
    res.json(result);
  })
);

router.post(
  '/:ticketId/release',
  getCurrentUser,
  asyncHandler(async (req, res) => {
    await ticketService.releaseTicketHold(req.params.ticketId);
    res.status(204).send();
  })
);

module.exports = router;
